src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
```

Comment:
I actually kind of prefer what @jeffmccune did here over what's currently in the code.  Less conditional logic and enforces consistency in the `rule-fn` signature.  MIght make this easier to maintain in the future if we were to add more `rule-fn` types as well.  I don't have a strong opinion on this, though, so if @nwolfe were to convince @jeffmccune to forgo this change, I'd be okay with that, too.